blinky/utils.py

```diff
@@ -129,3 +129,15 @@ def getchar(msg):
 		termios.tcsetattr(fd, termios.TCSADRAIN, old_settings)
 	print()
 	return ch
+
+
+def default_blinky_dir():
+	from xdg import BaseDirectory
+	# load_data_paths (generator) is empty if no data dirs exist
+	# so we can fall back to checking ~/.blinky
+	for xdg_dir in BaseDirectory.load_data_paths('blinky'):
+		return xdg_dir
+	# if the user does not have an existing ~/.blinky dir, start using the XDG default
```

Comment:
so you're suggesting to drop 138f and tell users "the first time you use <new version>" move your ~/.blinky to $XDG_… because if you fail to do so, blinky will create for you a second blinky dir and you'll have to merge by hand? (just cynically trying to anticipate the worst case). I guess rather than loading the backward compatible branch, 138 could be turned into a warning "you don't have an up to date config, but i saw there is something else you have". which keeps what blinky is actually doing minimal (no backwards compatibility there) but does give users some heuristic error message. What I'm uncertain about atm is, this diagnostics would only run when blinky is launched to actually do something. which is for many operations too late (i.e the split "here is your second config dir" happens).

I guess what I'm saying is "is your policy that users have to read the migration guide before going 'i need to install X right now and i got a new blinky through some automatic update'"?

(does any of what i wrote make sense or did i read too much into your comment?)

Reply:
It does. I was thinking to require a read of the migration guide as long as we are still in a 0.x-Release (wouldn't do this withinin a 1.x-branch, this would yield a 2.0). I would have thought about placing a post-upgrade-notification in the install-file, telling the user about the change so that everyone should see it (incidentially twice, once wen pacman installs the package and once when blinky requires review of said file).

However, thinking about it we could also keep a migration option here (I would go for a migration option, either manually or automatically, instead of backwards compatibility as the migration instructions are fairly straightforward) and remove it upon hitting 1.x or remove it after two releases.

What do you think is the best way going forward here? We could possibly outright refuse operation as long as ~/.blinky exists, to avoid creating that need-to-merge situation.

Reply:
Especially if the decision is made to separate cache and data directories, we could even provide a migration script (if we don't, it might actually not be worth it as it is a simple `mv` in that case).